RedHatInsights/rhsm-subscriptions
# SWATCH-2042: Send swatch-contracts HTTP access logs to Splunk

Jira issue: SWATCH-2042

## Description
Enable HTTP access logs for swatch contracts and create a special Splunk handler to set the `source-type` for those messages.

## Testing

### Setup
1.  Start a local Splunk container
    ```
    podman-compose -f config/splunk/docker-compose.yml up
    ```
1.  Start swatch-contracts configured to talk to that Splunk container.
    ```
    SERVER_PORT=8001 \                                     
    ENABLE_SPLUNK_HEC=true \
    SPLUNK_HEC_URL=https://localhost:8088 \
    SPLUNK_HEC_TOKEN=29fe2838-cab6-4d17-a392-37b7b8f41f75 \
    ./gradlew :swatch-contracts:quarkusDev
    ```

### Steps
1. Create some access log entries by going to the Quarkus dev console.  This will make a bunch of HTTP requests to load the JS and CSS.  Wait about 20 seconds (since Splunk HEC sends in batches)  **Note the time of day**.
    ```
    http --follow :8001/q/dev-ui
    ```
1.  Kill swatch-contracts and restart with Splunk disabled.
    ```
    SERVER_PORT=8001 ./gradlew :swatch-contracts:quarkusDev
    ```
1.  Make the same `http` request and again **note the time of day**.

### Verification

1.  Go to the Splunk console at `http://localhost:8000` and log in with `admin:admin123`.
1.  Search on `namespace="local"`
1.  For your first run of swatch-contracts, you'll see normal log messages with a source-type of `quarkus_service` and HTTP access log messages with a source-type of `access_combined` (since we're using the `combined` variant of the [Common Log Format](https://en.wikipedia.org/wiki/Common_Log_Format)
![Selection_020](https://github.com/RedHatInsights/rhsm-subscriptions/assets/17975/e7ed5714-96c8-4d88-abce-d5b6343a57f0)
1.  During the second-run of swatch-contracts, you should not see any events logged to the Splunk console, nor should you see any stacktraces or messages reading "Error while sending events to Splunk HEC".
